main.py

```diff
@@ -20,6 +20,7 @@ def request_as_string(request: Request, hostname: str, port: int) -> str:
                      f"Host: {hostname}:{port}",
                      *request.headers,
                      HEADER_SEPARATOR.decode(),
+                     request.body if request.body is not None else ""
```

Comment:
Remember how we had to look for the header Content-Length to be able to say where the HTTP message would end during the recev while block?

You need to do the same here, and provide a `Content-Length` if you send a body, so the server can also know where to stop.

Reply:
Amazing thanks for the tip ;)